--- origsrc/omake-0.9.8.6/lib/configure/Configure.om	2007-10-25 23:11:08.000000000 -0500
+++ src/omake-0.9.8.6/lib/configure/Configure.om	2011-12-15 22:46:07.601766200 -0600
@@ -266,7 +266,7 @@ public.VerboseCheckCHeader(files) =
 # \end{doc}
 #
 public.CheckCLib(libs, funs) =
-    CFLAGS += $(addprefix -l, $(libs))
+    LDFLAGS += $(addprefix -l, $(libs))
 
     return $(TryLinkC $"""
 #ifdef __cplusplus
